<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>R packages of Mirko Signorelli</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="publications.html">
    <span class="fa fa-book"></span>
     
    Publications
  </a>
</li>
<li>
  <a href="r.html">
    <span class="fa fa-laptop"></span>
     
    R packages
  </a>
</li>
<li>
  <a href="talks.html">
    <span class="fa fa-bullhorn"></span>
     
    Presentations
  </a>
</li>
<li>
  <a href="awards.html">
    <span class="fas fa fas fa-trophy"></span>
     
    Awards
  </a>
</li>
<li>
  <a href="teaching.html">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Teaching
  </a>
</li>
<li>
  <a href="contact.html">
    <span class="fa fa-info"></span>
     
    Contact
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://twitter.com/signormirko">
    <span class="fa fa-twitter"></span>
     
    
  </a>
</li>
<li>
  <a href="https://github.com/mirkosignorelli">
    <span class="fa fa-github"></span>
     
    
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R packages and contributions</h1>

</div>


<div id="overview" class="section level3">
<h3>Overview</h3>
<p>All of the statistical methods that I have developed so far are implemented using the <a href="https://www.r-project.org/">R programming language</a>. When time and funding allowed it, I have created R packages that make it easy to everyone to apply these methods; when this was not possible, I have uploaded the R scripts used to estimate a given model on <a href="https://github.com/mirkosignorelli">my github page</a>.</p>
<p>My involvement in the R community can be summarized as follows:</p>
<ul>
<li><p>I am the <strong>author and mantainer of the R packages neat and ptmixed</strong>;</p></li>
<li><p>I was one of the organizers of the European R Users Meeting 2020 (e-Rum2020) conference;</p></li>
<li><p>I contributed code to the R package EnrichmentBrowser.</p></li>
</ul>
<p>Below you can find more details about my contributions.</p>
</div>
<div id="r-package-neat" class="section level3">
<h3>R package neat</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/81393545-d662c680-9120-11ea-9963-a737f963bd8d.png" align="right" alt="" width="250" /></p>
<p><strong>NEAT</strong> is an acronym that stands for <em>Network Enrichment Analysis Test</em>, a test that can be used to assess relations between sets of nodes (typically genes) in a network.</p>
<p>The method is described in Signorelli, Vinciotti and Wit (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352, and it is implemented in the R package neat, available from CRAN.</p>
<p>If you want to learn more about NEAT, you can:</p>
<ul>
<li><p><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6">read the paper</a> that describes the method;</p></li>
<li><p>read the <a href="https://cran.r-project.org/web/packages/neat/vignettes/neat.html">vignette: An introduction to the R package neat</a>;</p></li>
<li><p>visit the <a href="https://cran.r-project.org/web/packages/neat/">package page on CRAN</a>;</p></li>
<li><p>check out the <a href="https://cran.r-project.org/web/packages/neat/neat.pdf">package manual</a>.</p></li>
</ul>
<p>News: since October 2020, the NEAT test can be computed also using the Bioconductor package EnrichmentBrowser.</p>
</div>
<div id="r-package-ptmixed" class="section level3">
<h3>R package ptmixed</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/84284787-f3821f00-ab3c-11ea-85a7-220982ce518c.png" align="right" alt="" width="250" /></p>
<p><strong>ptmixed</strong> is an abbreviation that stands for <em>Poisson-Tweedie generalized linear mixed model</em>, a mixed-effects model that has been developed to flexibly model longitudinal count data that feature overdispersion, zero inflation and/or heavy tails.</p>
<p>The model is described in Signorelli, M., Spitali, P., Tsonaka, R. (2020). Poisson-Tweedie mixed-effects model: a flexible approach for the analysis of longitudinal RNA-seq data. Statistical Modelling, 20 (1), 9-29, and it is implemented in the R package ptmixed, available from CRAN.</p>
<p>To find out more about ptmixed:</p>
<ul>
<li><p><a href="https://journals.sagepub.com/doi/10.1177/1471082X20936017">read the article</a> that describes the methodology behind ptmixed;</p></li>
<li><p>read the <a href="https://cran.r-project.org/web/packages/ptmixed/vignettes/Overview_functionalities_ptmixed.html">vignette: An introduction to the R package ptmixed</a>;</p></li>
<li><p>have a look at a short <a href="https://www.youtube.com/watch?v=79lUgThZ4HE&amp;t=738s">5 minute talk about ptmixed</a> that I presented at eRum2020;</p></li>
<li><p>visit the <a href="https://cran.r-project.org/web/packages/ptmixed">package page on CRAN</a>;</p></li>
<li><p>check out the <a href="https://cran.r-project.org/web/packages/ptmixed/ptmixed.pdf">package manual</a>.</p></li>
</ul>
</div>
<div id="european-r-users-meeting-2020-conference-e-rum2020" class="section level3">
<h3>European R Users Meeting 2020 conference (e-Rum2020)</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/84913063-626df380-b0ba-11ea-8781-dbff9962e2c0.png" align="right" alt="" width="250" /></p>
<p>I was one of the organizers of the <strong>European R Users Meeting 2020 conference</strong> (e-Rum2020), both as member of the <a href="https://2020.erum.io/about/organizers">organizing committee</a> and as responsible of the promotion of the conference.</p>
<p>e-Rum2020 was originally planned as a physical event to be held in Milan in May 2020. However, due to the COVID19 pandemic we decided to turn the event into a free virtual conference - <strong>the very first fully virtual R conference ever!</strong></p>
<p>Wanna find more about e-Rum2020?</p>
<ul>
<li><p>check out the <a href="https://2020.erum.io/">website of e-Rum2020</a>;</p></li>
<li><p>have a look at my <a href="https://github.com/mirkosignorelli/erum2020">curated list of e-Rum2020 resources</a>.</p></li>
</ul>
</div>
<div id="r-code-used-in-methodological-publications" class="section level3">
<h3>R code used in methodological publications</h3>
<ul>
<li><p><a href="https://github.com/mirkosignorelli/ptmixed">Code for Signorelli, Spitali and Tsonaka (2020)</a>, Statistical Modelling</p></li>
<li><p><a href="https://github.com/mirkosignorelli/clustering-populations-of-networks">Code for Signorelli and Wit (2020)</a>, Statistical Modelling</p></li>
<li><p><a href="https://github.com/mirkosignorelli/penalized-blockmodels">Code for Signorelli and Wit (2018)</a>, JRSS-C</p></li>
<li><p><a href="https://github.com/mirkosignorelli/neat">Code for Signorelli, Vinciotti and Wit (2016)</a>, BMC Bioinformatics</p></li>
</ul>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
