<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>R packages of Mirko Signorelli</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet" />
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XZHFRKZJ8Y"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-XZHFRKZJ8Y');
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="publications.html">
    <span class="fa fa-book"></span>
     
    Publications
  </a>
</li>
<li>
  <a href="r.html">
    <span class="fa fa-laptop"></span>
     
    R packages
  </a>
</li>
<li>
  <a href="teaching.html">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Teaching
  </a>
</li>
<li>
  <a href="students.html">
    <span class="fa fa-user-graduate"></span>
     
    Students
  </a>
</li>
<li>
  <a href="awards.html">
    <span class="fas fa-trophy"></span>
     
    Awards
  </a>
</li>
<li>
  <a href="talks.html">
    <span class="fa fa-bullhorn"></span>
     
    Presentations
  </a>
</li>
<li>
  <a href="contact.html">
    <span class="fa fa-info"></span>
     
    Contact
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://twitter.com/signormirko">
    <span class="fab fa-twitter"></span>
     
    
  </a>
</li>
<li>
  <a href="https://github.com/mirkosignorelli">
    <span class="fab fa-github"></span>
     
    
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">R packages and contributions</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#overview" id="toc-overview">Overview</a></li>
<li><a href="#r-package-neat" id="toc-r-package-neat">R package
neat</a></li>
<li><a href="#r-package-pencal" id="toc-r-package-pencal">R package
pencal</a></li>
<li><a href="#r-package-ptmixed" id="toc-r-package-ptmixed">R package
ptmixed</a></li>
<li><a href="#r-package-success" id="toc-r-package-success">R package
success</a></li>
<li><a href="#european-r-users-meeting-2020-conference-e-rum2020"
id="toc-european-r-users-meeting-2020-conference-e-rum2020">European R
Users Meeting 2020 conference (e-Rum2020)</a></li>
</ul>
</div>

<div id="overview" class="section level3">
<h3>Overview</h3>
<p>When developing new statistical methods, I usually create
<code>R</code> packages that make it easier to apply them. When this is
not possible due to time or funding constraints, I upload the
<code>R</code> code I used on <a
href="https://github.com/mirkosignorelli">my github page</a>.</p>
<p>My involvement in the <code>R</code> community so far can be
summarized as follows:</p>
<ul>
<li><p>I am the author and mantainer of the <strong><code>R</code>
packages <code>neat</code>, <code>ptmixed</code> and
<code>pencal</code></strong>;</p></li>
<li><p>I contributed to the development of the <strong><code>R</code>
package <code>success</code></strong> (developed and mantained by <a
href="https://github.com/d-gomon">Daniel Gomon</a>);</p></li>
<li><p>I was one of the organizers of the <strong>European R Users
Meeting 2020 (e-Rum2020) conference</strong>;</p></li>
<li><p>I made minor contributions to the <code>R</code> packages
<code>EnrichmentBrowser</code> and <code>markovchain</code>.</p></li>
</ul>
<p>Below you can find more details about some of these
contributions.</p>
</div>
<div id="r-package-neat" class="section level3">
<h3>R package neat</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/81393545-d662c680-9120-11ea-9963-a737f963bd8d.png" align="right" alt="" width="250" /></p>
<p><strong>NEAT</strong> is an acronym that stands for <em>Network
Enrichment Analysis Test</em>, a test that can be used to assess
relations between sets of nodes (typically genes) in a network.</p>
<p>The method is described in Signorelli, Vinciotti and Wit (2016).
NEAT: an efficient network enrichment analysis test. <em>BMC
Bioinformatics</em>, 17:352, and it is implemented in the <code>R</code>
package <code>neat</code>, available from CRAN.</p>
<p>If you want to learn more about NEAT, you can:</p>
<ul>
<li><p><a
href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6">read
the paper</a> that describes the method;</p></li>
<li><p>read the <a
href="https://cran.r-project.org/web/packages/neat/vignettes/neat.html">vignette:
An introduction to the <code>R</code> package
<code>neat</code></a>;</p></li>
<li><p>visit the <a
href="https://cran.r-project.org/web/packages/neat/">package page on
CRAN</a>, and check out the <a
href="https://cran.r-project.org/web/packages/neat/neat.pdf">package
manual</a>.</p></li>
</ul>
<p>News: since November 2020, the NEAT test can be computed also using
the Bioconductor package <code>EnrichmentBrowser</code>.</p>
</div>
<div id="r-package-pencal" class="section level3">
<h3>R package pencal</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/162180793-072613f0-a93e-4ef6-b0c4-b8d8a45d770a.png" align="right" alt="" width="250" /></p>
<p><strong><code>pencal</code></strong> is an abbreviation that stands
for <em>Penalized Regression Calibration (PRC)</em>, a method that can
be used to predict a survival outcome given a set of predictors that are
high-dimensional and longitudinally measured.</p>
<p>The method is described in Signorelli <em>et al.</em> (2021).
Penalized regression calibration: a method for the prediction of
survival outcomes using complex longitudinal and high-dimensional data.
<em>Statistics in Medicine</em>, 40 (27), 6178-6196, and it is
implemented in the <code>R</code> package <code>pencal</code>, available
from CRAN.</p>
<p>To find out more about PRC and <code>pencal</code>, check out the
following resources:</p>
<ul>
<li><p><a href="https://doi.org/10.1002/sim.9178">read the article</a>
that describes PRC;</p></li>
<li><p>read the <a
href="https://cran.r-project.org/web/packages/pencal/vignettes/pencal-vignette.html">vignette:
An introduction to the <code>R</code> package
<code>pencal</code></a>;</p></li>
<li><p>visit the <a
href="https://cran.r-project.org/web/packages/pencal">package page on
CRAN</a>, and check out the <a
href="https://cran.r-project.org/web/packages/pencal/pencal.pdf">package
manual</a>.</p></li>
</ul>
</div>
<div id="r-package-ptmixed" class="section level3">
<h3>R package ptmixed</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/84284787-f3821f00-ab3c-11ea-85a7-220982ce518c.png" align="right" alt="" width="250" /></p>
<p><strong><code>ptmixed</code></strong> is an abbreviation that stands
for <em>Poisson-Tweedie generalized linear mixed model</em>, a
mixed-effects model that has been developed to flexibly model
longitudinal count data that feature overdispersion, zero inflation
and/or heavy tails.</p>
<p>The model is described in Signorelli, M., Spitali, P., Tsonaka, R.
(2021). Poisson-Tweedie mixed-effects model: a flexible approach for the
analysis of longitudinal RNA-seq data. <em>Statistical Modelling</em>,
21 (6), 520-545, and it is implemented in the <code>R</code> package
<code>ptmixed</code>, available from CRAN.</p>
<p>To find out more about <code>ptmixed</code>:</p>
<ul>
<li><p><a
href="https://journals.sagepub.com/doi/10.1177/1471082X20936017">read
the article</a> that describes the methodology behind ptmixed;</p></li>
<li><p>read the <a
href="https://cran.r-project.org/web/packages/ptmixed/vignettes/Overview_functionalities_ptmixed.html">vignette:
An introduction to the <code>R</code> package
<code>ptmixed</code></a>;</p></li>
<li><p>have a look at a short <a
href="https://www.youtube.com/watch?v=79lUgThZ4HE&amp;t=738s">5 minute
talk about <code>ptmixed</code></a> that I presented at
eRum2020;</p></li>
<li><p>visit the <a
href="https://cran.r-project.org/web/packages/ptmixed">package page on
CRAN</a>, and check out the <a
href="https://cran.r-project.org/web/packages/ptmixed/ptmixed.pdf">package
manual</a>.</p></li>
</ul>
</div>
<div id="r-package-success" class="section level3">
<h3>R package success</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/178921364-986c04b3-71ec-4cbe-8e46-2c36ede13a7c.png" align="right" alt="" width="250" /></p>
<p><strong><code>success</code></strong> (short for: Survival Control
Charts Estimation Software) is an <code>R</code> package that implements
methods to compute and visualize the following charts for quality
control of survival outcomes:</p>
<ol style="list-style-type: decimal">
<li>funnel plot;</li>
<li>Bernoulli CUSUM;</li>
<li>Biswas &amp; Kalbfleischâ€™s CUSUM (BK-CUSUM);</li>
<li>Continuous Time Generalized Rapid Response CUSUM (CGR-CUSUM).</li>
</ol>
<p>Note that the funnel plot and Bernoulli CUSUM require a
dychotomization of the survival outcome into a binary outcome, whereas
the CGR-CUSUM and BK-CUSUM can properly handle the survival outcome.</p>
<p>To find out more about <code>success</code>:</p>
<ul>
<li><p>visit the <a
href="https://cran.r-project.org/web/packages/success">package page on
CRAN</a>;</p></li>
<li><p>check out this <a
href="https://arxiv.org/abs/2302.07658">article</a> that illustrates how
the package works and what you can do with it;</p></li>
<li><p>visit the <a
href="https://cran.r-project.org/web/packages/success">package page on
CRAN</a>, and check out the <a
href="https://cran.r-project.org/web/packages/success/success.pdf">package
manual</a>.</p></li>
</ul>
</div>
<div id="european-r-users-meeting-2020-conference-e-rum2020"
class="section level3">
<h3>European R Users Meeting 2020 conference (e-Rum2020)</h3>
<p><img src="https://user-images.githubusercontent.com/20061736/84913063-626df380-b0ba-11ea-8781-dbff9962e2c0.png" align="right" alt="" width="250" /></p>
<p>I was one of the organizers of the <strong>European R Users Meeting
2020 conference</strong> (e-Rum2020), both as member of the <a
href="https://2020.erum.io/about/organizers">organizing committee</a>
and as responsible of the promotion of the conference.</p>
<p>e-Rum2020 was originally planned as a physical event to be held in
Milan in May 2020. However, due to the COVID19 pandemic we decided to
turn the event into a free virtual conference - <strong>the very first
fully virtual R conference ever!</strong></p>
<p>If you would like to know more about e-Rum2020, you can:</p>
<ul>
<li><p>read the article <a
href="https://journal.r-project.org/archive/2020-2/erum.pdf">e-Rum2020:
how we turned a physical conference into a successful virtual event</a>,
published in <em>The R Journal</em>, in which we summarized our
adventure as conference organizers in the middle of a pandemic
outbreak;</p></li>
<li><p>check out the <a href="https://2020.erum.io/">website of
e-Rum2020</a>;</p></li>
<li><p>have a look at my <a
href="https://github.com/mirkosignorelli/erum2020">curated list of
e-Rum2020 resources</a>.</p></li>
</ul>
<!---
### R code used in methodological publications

* [Code for Signorelli et al. (2021)](https://github.com/mirkosignorelli/pencal), Statistics in Medicine

* [Code for Signorelli and Cutillo (2021)](https://github.com/mirkosignorelli/csv), Computational Statistics

* [Code for Signorelli, Spitali and Tsonaka (2021)](https://github.com/mirkosignorelli/ptmixed), Statistical Modelling

* [Code for Signorelli and Wit (2020)](https://github.com/mirkosignorelli/clustering-populations-of-networks), Statistical Modelling

* [Code for Signorelli and Wit (2018)](https://github.com/mirkosignorelli/penalized-blockmodels), JRSS-C

* [Code for Signorelli, Vinciotti and Wit (2016)](https://github.com/mirkosignorelli/neat), BMC Bioinformatics
-->
<!-- the part below is a fix to the menu not displaying properly
on the mobile version -->
<script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
<script src="https://cdn.jsdelivr.net/npm/popper.js@1.14.7/dist/umd/popper.min.js" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" crossorigin="anonymous"></script>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
