---
title: "R packages and contributions"
pagetitle: "R packages of Mirko Signorelli"
output:
  html_document:
    toc: true
---

### Overview

When developing new statistical methods, I usually create `R` packages that make it easier to apply them. When this is not possible due to time or funding constraints, I upload the `R` code I used on [my github page](https://github.com/mirkosignorelli).

My involvement in the `R` community so far can be summarized as follows:

* I am the author and mantainer of the **`R` packages `neat`, `ptmixed` and `pencal`**;

* I contributed to the development of the **`R` package `success`** (developed and mantained by [Daniel Gomon](https://github.com/d-gomon));

* I was one of the organizers of the **European R Users Meeting 2020 (e-Rum2020) conference**;

* I made minor contributions to the `R` packages `EnrichmentBrowser` and `markovchain`.

Below you can find more details about some of these contributions.

### R package neat

<img src="https://user-images.githubusercontent.com/20061736/81393545-d662c680-9120-11ea-9963-a737f963bd8d.png" align="right" alt="" width="250" />

**NEAT** is an acronym that stands for *Network Enrichment Analysis Test*, a test that can be used to assess relations between sets of nodes (typically genes) in a network.

The method is described in Signorelli, Vinciotti and Wit (2016). NEAT: an efficient network enrichment analysis test. *BMC Bioinformatics*, 17:352, and it is implemented in the `R` package `neat`, available from CRAN.

If you want to learn more about NEAT, you can:

* [read the paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6) that describes the method;

* read the [vignette: An introduction to the `R` package `neat`](https://cran.r-project.org/web/packages/neat/vignettes/neat.html);

* visit the [package page on CRAN](https://cran.r-project.org/web/packages/neat/), and check out the [package manual](https://cran.r-project.org/web/packages/neat/neat.pdf).

News: since November 2020, the NEAT test can be computed also using the Bioconductor package `EnrichmentBrowser`.

### R package pencal

<img src="https://user-images.githubusercontent.com/20061736/162180793-072613f0-a93e-4ef6-b0c4-b8d8a45d770a.png" align="right" alt="" width="250" />

**`pencal`** is an abbreviation that stands for *Penalized Regression Calibration (PRC)*, a method that can be used to predict a survival outcome given a set of predictors that are high-dimensional and longitudinally measured. PRC is implemented in the `R` package `pencal`, available from CRAN.

PRC was first described in Signorelli *et al.* (2021). Penalized regression calibration: a method for the prediction of survival outcomes using complex longitudinal and high-dimensional data. *Statistics in Medicine*, 40 (27), 6178-6196. A detailed description of how to use PRC to do dynamic prediction of survival can be found in Signorelli (in review). `pencal`: an `R` Package for the Dynamic Prediction of Survival with Many Longitudinal Predictors.

To find out more about PRC and `pencal`, check out the following resources:

* read the [Statistics in Medicine article](https://doi.org/10.1002/sim.9178) where PRC was proposed;

* read the article [pencal: an R Package for the Dynamic Prediction of Survival with Many Longitudinal Predictors](https://cran.r-project.org/web/packages/pencal/vignettes/vignette.pdf);

* visit the [package page on CRAN](https://cran.r-project.org/web/packages/pencal), and check out the [package manual](https://cran.r-project.org/web/packages/pencal/pencal.pdf).

### R package ptmixed

<img src="https://user-images.githubusercontent.com/20061736/84284787-f3821f00-ab3c-11ea-85a7-220982ce518c.png" align="right" alt="" width="250" />

**`ptmixed`** is an abbreviation that stands for *Poisson-Tweedie generalized linear mixed model*, a mixed-effects model that has been developed to flexibly model longitudinal count data that feature overdispersion, zero inflation and/or heavy tails. 

The model is described in Signorelli, M., Spitali, P., Tsonaka, R. (2021). Poisson-Tweedie mixed-effects model: a flexible approach for the analysis of longitudinal RNA-seq data. *Statistical Modelling*, 21 (6), 520-545, and it is implemented in the `R` package `ptmixed`, available from CRAN.

To find out more about `ptmixed`:

* [read the article](https://journals.sagepub.com/doi/10.1177/1471082X20936017) that describes the methodology behind ptmixed;

* read the [vignette: An introduction to the `R` package `ptmixed`](https://cran.r-project.org/web/packages/ptmixed/vignettes/Overview_functionalities_ptmixed.html);

* have a look at a short [5 minute talk about `ptmixed`](https://www.youtube.com/watch?v=79lUgThZ4HE&t=738s) that I presented at eRum2020;

* visit the [package page on CRAN](https://cran.r-project.org/web/packages/ptmixed), and check out the [package manual](https://cran.r-project.org/web/packages/ptmixed/ptmixed.pdf).

### R package success

<img src="https://user-images.githubusercontent.com/20061736/178921364-986c04b3-71ec-4cbe-8e46-2c36ede13a7c.png" align="right" alt="" width="250" />

**`success`** (short for: Survival Control Charts Estimation Software) is an `R` package that implements methods to compute and visualize the following charts for quality control of survival outcomes:

1. funnel plot;
2. Bernoulli CUSUM;
3. Biswas & Kalbfleisch's CUSUM (BK-CUSUM);
4. Continuous Time Generalized Rapid Response CUSUM (CGR-CUSUM).

Note that the funnel plot and Bernoulli CUSUM require a dychotomization of the survival outcome into a binary outcome, whereas the CGR-CUSUM and BK-CUSUM can properly handle the survival outcome.

The `R` package is described in: Gomon, D., Fiocco, M., Putter, H., **Signorelli, M.** (2023). [SUrvival Control Chart EStimation Software in `R`: the `success` package](https://doi.org/10.32614/RJ-2023-095). *The R Journal*, 15 (4), 270-291.

To find out more about `success`:

* read the article [SUrvival Control Chart EStimation Software in `R`: the `success` package](https://doi.org/10.32614/RJ-2023-095);
* visit the [package page on CRAN](https://cran.r-project.org/web/packages/success), and check out the [package manual](https://cran.r-project.org/web/packages/success/success.pdf).

### European R Users Meeting 2020 conference (e-Rum2020)

<img src="https://user-images.githubusercontent.com/20061736/84913063-626df380-b0ba-11ea-8781-dbff9962e2c0.png" align="right" alt="" width="250" />

I was one of the organizers of the **European R Users Meeting 2020 conference** (e-Rum2020), both as member of the [organizing committee](https://2020.erum.io/about/organizers) and as responsible of the promotion of the conference. 

e-Rum2020 was originally planned as a physical event to be held in Milan in May 2020. However, due to the COVID19 pandemic we decided to turn the event into a free virtual conference - **the very first fully virtual R conference ever!**

If you would like to know more about e-Rum2020, you can:

* read the article [e-Rum2020: how we turned a physical conference into a successful virtual event](https://journal.r-project.org/archive/2020-2/erum.pdf), published in *The R Journal*, in which we summarized our adventure as conference organizers in the middle of a pandemic outbreak;

* check out the [website of e-Rum2020](https://2020.erum.io/);

* have a look at my [curated list of e-Rum2020 resources](https://github.com/mirkosignorelli/erum2020).

<!---
### R code used in methodological publications

* [Code for Signorelli et al. (2021)](https://github.com/mirkosignorelli/pencal), Statistics in Medicine

* [Code for Signorelli and Cutillo (2021)](https://github.com/mirkosignorelli/csv), Computational Statistics

* [Code for Signorelli, Spitali and Tsonaka (2021)](https://github.com/mirkosignorelli/ptmixed), Statistical Modelling

* [Code for Signorelli and Wit (2020)](https://github.com/mirkosignorelli/clustering-populations-of-networks), Statistical Modelling

* [Code for Signorelli and Wit (2018)](https://github.com/mirkosignorelli/penalized-blockmodels), JRSS-C

* [Code for Signorelli, Vinciotti and Wit (2016)](https://github.com/mirkosignorelli/neat), BMC Bioinformatics
-->


<!-- the part below is a fix to the menu not displaying properly
on the mobile version -->
<script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
<script src="https://cdn.jsdelivr.net/npm/popper.js@1.14.7/dist/umd/popper.min.js" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" crossorigin="anonymous"></script>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
