---
title: "R packages and contributions"
pagetitle: "R packages of Mirko Signorelli"
---

### Overview

All of the statistical methods that I have developed so far are implemented using the [R programming language](https://www.r-project.org/). When time and funding allowed it, I have created R packages that make it easy to everyone to apply these methods; when this was not possible, I have uploaded the R scripts used to estimate a given model on [my github page](https://github.com/mirkosignorelli).

My involvement in the R community can be summarized as follows:

* I am the **author and mantainer of the R packages `neat` and `ptmixed`**;

* I was one of the organizers of the European R Users Meeting 2020 (e-Rum2020) conference;

* I contributed code to the R package `EnrichmentBrowser`.

Below you can find more details about my contributions.

### R package neat

<img src="https://user-images.githubusercontent.com/20061736/81393545-d662c680-9120-11ea-9963-a737f963bd8d.png" align="right" alt="" width="250" />

**NEAT** is an acronym that stands for *Network Enrichment Analysis Test*, a test that can be used to assess relations between sets of nodes (typically genes) in a network.

The method is described in Signorelli, Vinciotti and Wit (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352, and it is implemented in the R package `neat`, available from CRAN.

If you want to learn more about NEAT, you can:

* [read the paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6) that describes the method;

* read the [vignette: An introduction to the R package neat](https://cran.r-project.org/web/packages/neat/vignettes/neat.html);

* visit the [package page on CRAN](https://cran.r-project.org/web/packages/neat/);

* check out the [package manual](https://cran.r-project.org/web/packages/neat/neat.pdf).

News: since November 2020, the NEAT test can be computed also using the Bioconductor package `EnrichmentBrowser`.

### R package ptmixed

<img src="https://user-images.githubusercontent.com/20061736/84284787-f3821f00-ab3c-11ea-85a7-220982ce518c.png" align="right" alt="" width="250" />

**ptmixed** is an abbreviation that stands for *Poisson-Tweedie generalized linear mixed model*, a mixed-effects model that has been developed to flexibly model longitudinal count data that feature overdispersion, zero inflation and/or heavy tails. 

The model is described in Signorelli, M., Spitali, P., Tsonaka, R. (2020). Poisson-Tweedie mixed-effects model: a flexible approach for the analysis of longitudinal RNA-seq data. Statistical Modelling, 20 (1), 9-29, and it is implemented in the R package `ptmixed`, available from CRAN.

To find out more about ptmixed:

* [read the article](https://journals.sagepub.com/doi/10.1177/1471082X20936017) that describes the methodology behind ptmixed;

* read the [vignette: An introduction to the R package ptmixed](https://cran.r-project.org/web/packages/ptmixed/vignettes/Overview_functionalities_ptmixed.html);

* have a look at a short [5 minute talk about ptmixed](https://www.youtube.com/watch?v=79lUgThZ4HE&t=738s) that I presented at eRum2020;

* visit the [package page on CRAN](https://cran.r-project.org/web/packages/ptmixed);

* check out the [package manual](https://cran.r-project.org/web/packages/ptmixed/ptmixed.pdf).

### European R Users Meeting 2020 conference (e-Rum2020)

<img src="https://user-images.githubusercontent.com/20061736/84913063-626df380-b0ba-11ea-8781-dbff9962e2c0.png" align="right" alt="" width="250" />

I was one of the organizers of the **European R Users Meeting 2020 conference** (e-Rum2020), both as member of the [organizing committee](https://2020.erum.io/about/organizers) and as responsible of the promotion of the conference. 

e-Rum2020 was originally planned as a physical event to be held in Milan in May 2020. However, due to the COVID19 pandemic we decided to turn the event into a free virtual conference - **the very first fully virtual R conference ever!**

Wanna find more about e-Rum2020?

* check out the [website of e-Rum2020](https://2020.erum.io/);

* have a look at my [curated list of e-Rum2020 resources](https://github.com/mirkosignorelli/erum2020).

### R code used in methodological publications

* [Code for Signorelli, Spitali and Tsonaka (2020)](https://github.com/mirkosignorelli/ptmixed), Statistical Modelling

* [Code for Signorelli and Wit (2020)](https://github.com/mirkosignorelli/clustering-populations-of-networks), Statistical Modelling

* [Code for Signorelli and Wit (2018)](https://github.com/mirkosignorelli/penalized-blockmodels), JRSS-C

* [Code for Signorelli, Vinciotti and Wit (2016)](https://github.com/mirkosignorelli/neat), BMC Bioinformatics
